Plugin/Framework/App/Response/Http.php

```diff
@@ -0,0 +1,172 @@
+<?php declare(strict_types=1);
+
+namespace MageOS\ThemeOptimization\Plugin\Framework\App\Response;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\Request\Http as HttpRequest;
+use Magento\PageCache\Model\Config;
+use Magento\Store\Model\ScopeInterface;
+
+/**
+ * Plugin to modify cache headers for BFCache functionality
+ */
+class Http
+{
+    /** Configuration paths */
+    const XML_PATH_ENABLE = 'system/bfcache/general/enable';
+    const XML_PATH_BLACK_LIST_URLS = 'system/bfcache/scope/black_list_urls';
+
+    /** @var Config */
+    private $config;
+
+    /** @var ScopeConfigInterface */
+    private $scopeConfig;
+
+    /** @var HttpRequest */
+    private $request;
+
+    /** @var bool */
+    private $isRequestCacheable = false;
+
+    /**
+     * @param Config $config
+     * @param ScopeConfigInterface $scopeConfig
+     * @param HttpRequest $request
+     */
+    public function __construct(
+        Config $config,
+        ScopeConfigInterface $scopeConfig,
+        HttpRequest $request
+    ) {
+        $this->config = $config;
+        $this->scopeConfig = $scopeConfig;
+        $this->request = $request;
+    }
+
+    /**
+     * Intercept before setting no-cache headers to determine if request is cacheable
+     *
+     * @param \Magento\Framework\App\Response\Http $subject
+     * @return void
+     */
+    public function beforeSetNoCacheHeaders(\Magento\Framework\App\Response\Http $subject): void
+    {
+        if ($this->config->getType() !== Config::BUILT_IN || !$this->isEnabled()) {
+            return;
+        }
+
+        $cacheControlHeader = $subject->getHeader('Cache-Control');
+        if (!$cacheControlHeader) {
+            return;
+        }
+
+        $cacheControl = $cacheControlHeader->getFieldValue();
+        $requestURI = ltrim($this->request->getRequestURI(), '/');
+        
```

Comment:
```suggestion

```

Reply:
we already update this in the latest commit